renderer/src/hooks/StationWallet.tsx

```diff
@@ -59,38 +89,58 @@ const useWallet = (): Wallet => {
 
   useEffect(() => {
     const loadStoredInfo = async () => {
-      setWalletTransactions(await getStationWalletTransactionsHistory())
+      const { processing, confirmed } = splitWalletTransactions(
+        await getStationWalletTransactionsHistory()
+      )
+      setTransactions(processing, confirmed)
     }
     loadStoredInfo()
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
+  }, [setTransactions])
 
   useEffect(() => {
-    const updateWalletTransactionsArray = (transactions: FILTransaction[]) => {
-      const [newCurrentTransaction, ...confirmedTransactions] = transactions
-      if (newCurrentTransaction.status === 'processing' || (currentTransaction && +currentTransaction.timestamp === +newCurrentTransaction.timestamp)) {
-        setCurrentTransaction(newCurrentTransaction)
-        if (newCurrentTransaction.status !== 'processing') { setTimeout(() => { setWalletTransactions(transactions); setCurrentTransaction(undefined) }, 6000) }
-        setWalletTransactions(confirmedTransactions)
-      } else {
-        setWalletTransactions(transactions)
-      }
+    const updateWalletTransactionsArray = (transactions: (FILTransaction|FILTransactionProcessing)[]) => {
+      const { processing, confirmed } = splitWalletTransactions(transactions)
+      setTransactions(processing, confirmed)
     }
 
     const unsubscribeOnTransactionUpdate = window.electron.stationEvents.onTransactionUpdate(updateWalletTransactionsArray)
     return () => {
       unsubscribeOnTransactionUpdate()
     }
-  }, [currentTransaction])
+  }, [setTransactions])
 
   useEffect(() => {
-    const unsubscribeOnBalanceUpdate = window.electron.stationEvents.onBalanceUpdate(setWalletBalance)
+    const updateWalletBalance = (balance: string) => {
+      setWalletBalance(balance)
+    }
+
+    const unsubscribeOnBalanceUpdate = window.electron.stationEvents.onBalanceUpdate(updateWalletBalance)
+    return () => {
+      unsubscribeOnBalanceUpdate()
+    }
+  }, [])
+
+  useEffect(() => {
+    const unsubscribeOnBalanceUpdate = window.electron.stationEvents.onBalanceUpdate(balance => setWalletBalance(balance))
     return () => {
       unsubscribeOnBalanceUpdate()
     }
   }, [walletBalance])
 
-  return { stationAddress, destinationFilAddress, walletBalance, walletTransactions, editDestinationAddress, currentTransaction, dismissCurrentTransaction }
+  return { stationAddress, destinationFilAddress, walletBalance, walletTransactions, editDestinationAddress, processingTransaction, dismissCurrentTransaction }
+}
+
+interface SplitTransactions {
+  processing: FILTransactionProcessing | undefined,
+  confirmed: FILTransaction[]
+}
+
+const splitWalletTransactions = (transactions: (FILTransaction|FILTransactionProcessing)[]): SplitTransactions => {
+  const processing: FILTransactionProcessing | null =
+  transactions.filter(tx => tx.status === 'processing')[0] as FILTransactionProcessing
```

Comment:
I would consider implementing a [Type Guard function](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) to discriminate between different transaction types.

```ts
//
// inside typings
//

export function isFILTransactionProcessing(tx: FILTransaction | FILTransactionProcessing): tx is FILTransactionProcessing {
  return tx.status === 'processing'
}

export function isFILTransactionConfirmed(tx: FILTransaction | FILTransactionProcessing): tx is FILTransaction {
  return tx.status !== 'processing'
}

//
// usage here
//

const processing = transactions.filter(isFILTransactionProcessing)[0]
// btw, is it better to use transactions.find() instead?
const processing = transactions.find(isFILTransactionProcessing)

const confirmed = transactions.find(isFILTransactionConfirmed)
```

Reply:
all good points! I've implemented it all :)